kernel/src/table_properties/deserialize.rs

```diff
@@ -58,81 +56,83 @@ fn try_parse(props: &mut TableProperties, k: &str, v: &str) -> Option<()> {
+            props.set_transaction_retention_duration = parse_interval(v)?
         }
+        "delta.targetFileSize" => props.target_file_size = parse_positive_int(v)?,
+        "delta.tuneFileSizesForRewrites" => props.tune_file_sizes_for_rewrites = parse_bool(v)?,
         "delta.checkpointPolicy" => props.checkpoint_policy = CheckpointPolicy::try_from(v).ok(),
-        "delta.enableRowTracking" => props.enable_row_tracking = Some(parse_bool(v)?),
-        "delta.enableInCommitTimestamps" => {
-            props.enable_in_commit_timestamps = Some(parse_bool(v)?)
-        }
+        "delta.enableRowTracking" => props.enable_row_tracking = parse_bool(v)?,
+        "delta.enableInCommitTimestamps" => props.enable_in_commit_timestamps = parse_bool(v)?,
         "delta.inCommitTimestampEnablementVersion" => {
-            props.in_commit_timestamp_enablement_version = Some(parse_non_negative(v)?)
+            props.in_commit_timestamp_enablement_version = parse_non_negative(v)?
         }
         "delta.inCommitTimestampEnablementTimestamp" => {
-            props.in_commit_timestamp_enablement_timestamp = Some(parse_non_negative(v)?)
+            props.in_commit_timestamp_enablement_timestamp = parse_non_negative(v)?
         }
-        _ => return None,
+        _ => return Err("Invalid value"),
     }
-    Some(())
+    Ok(())
 }
 
 /// Deserialize a string representing a positive (> 0) integer into an `Option<u64>`. Returns `Some`
 /// if successfully parses, and `None` otherwise.
-pub(crate) fn parse_positive_int(s: &str) -> Option<NonZero<u64>> {
+pub(crate) fn parse_positive_int(s: &str) -> Result<Option<NonZero<u64>>, &'static str> {
     // parse as non-negative and verify the result is non-zero
-    NonZero::new(parse_non_negative(s)?)
+    parse_non_negative(s).map(|maybe_num| maybe_num.and_then(NonZero::<u64>::new))
 }
 
 /// Deserialize a string representing a non-negative integer into an `Option<u64>`. Returns `Some` if
 /// successfully parses, and `None` otherwise.
-pub(crate) fn parse_non_negative<T>(s: &str) -> Option<T>
+pub(crate) fn parse_non_negative<T>(s: &str) -> Result<Option<T>, &'static str>
 where
     i64: TryInto<T>,
 {
     // parse to i64 since java doesn't even allow u64
-    let n: i64 = s.parse().ok().filter(|&i| i >= 0)?;
-    n.try_into().ok()
+    Ok(s.parse::<i64>()
+        .ok()
+        .filter(|&i| i >= 0)
+        .and_then(|value| value.try_into().ok()))
+    //.map(|value| Some(value))
+    //.ok_or("Invalid value")
 }
 
 /// Deserialize a string representing a boolean into an `Option<bool>`. Returns `Some` if
 /// successfully parses, and `None` otherwise.
-pub(crate) fn parse_bool(s: &str) -> Option<bool> {
+pub(crate) fn parse_bool(s: &str) -> Result<Option<bool>, &'static str> {
     match s {
-        "true" => Some(true),
-        "false" => Some(false),
-        _ => None,
+        "true" => Ok(Some(true)),
+        "false" => Ok(Some(false)),
+        _ => Err("Invalid value"),
```

Comment:
Is it meaningful to return Optional? maybe just bool with false as default value?